crypto/kms/pubkeystore.go

```diff
@@ -0,0 +1,189 @@
+/*
+ * Copyright 2018 The ThunderDB Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package kms
+
+import (
+	"errors"
+
+	"encoding/hex"
+
+	"sync"
+
+	log "github.com/sirupsen/logrus"
+
+	ec "github.com/btcsuite/btcd/btcec"
+	"github.com/coreos/bbolt"
+	"github.com/thunderdb/ThunderDB/crypto/hash"
+	mine "github.com/thunderdb/ThunderDB/pow/cpuminer"
+	"github.com/thunderdb/ThunderDB/proto"
+)
+
+// PublicKeyStore holds db and bucket name
+type PublicKeyStore struct {
+	db     *bolt.DB
+	bucket []byte
+}
+
+const (
+	// kmsBucketName is the boltdb bucket name
+	kmsBucketName = "kms"
+)
+
+var (
+	// pks holds the singleton instance
+	pks     *PublicKeyStore
+	pksOnce sync.Once
+)
+
+var (
+	// BPPublicKey is the public key of Block Producer
```

Comment:
**[golint]** <sub>reported by [reviewdog](https://github.com/haya14busa/reviewdog) :dog:</sub>
comment on exported var BPPublicKeyStr should be of the form "BPPublicKeyStr ..."